### PR TITLE
remove superfluous backticks across pkg documentation

### PR DESCRIPTION
**Pull request by luisDVA**
addresses issue #1134 
removed backticks around package names per the tidy style guide. In few cases I edited the text to say "the parsnip package..." for disambiguation. This backtick audit considered downlit linking and I left some package names with backticks intentionally. (I ran some of these edge cases by the posit crew during dev day).

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 * `fit_xy()` currently raises an error for `gen_additive_mod()` model specifications as the default engine (`"mgcv"`) specifies smoothing terms in model formulas. However, some engines specify smooths via additional arguments, in which case the restriction on `fit_xy()` is excessive. parsnip will now only raise an error when fitting a `gen_additive_mod()` with `fit_xy()` when using the `"mgcv"` engine (#775).
 
-* Aligned `null_model()` with other model types; the model type now has an engine argument that defaults to `"parsnip"` and is checked with the same machinery that checks other model types in the package (#1083).
+* Aligned `null_model()` with other model types; the model type now has an engine argument that defaults to parsnip and is checked with the same machinery that checks other model types in the package (#1083).
```
Comment (simonpcouch) [inline NEWS.md:6]:
```suggestion
* Aligned `null_model()` with other model types; the model type now has an engine argument that defaults to `"parsnip"` and is checked with the same machinery that checks other model types in the package (#1083).
```

Let's keep this one as-is, as it does indeed refer to the character string `"parsnip"`!

File: NEWS.md [hunk 2]
```diff
@@ -121,7 +121,7 @@ This release of parsnip contains a number of new features and bug fixes, accompa
 
 # parsnip 1.0.3
 
-* Adds documentation and tuning infrastructure for the new `flexsurvspline` engine for the `survival_reg()` model specification from the `censored` package (@mattwarkentin, #831).
```
Comment (simonpcouch) [inline NEWS.md:124]: Generally, we try not to touch NEWS entries once they've made it to CRAN. Could you revert the changes to this file?

**Review by simonpcouch — COMMENTED** [review]
Wonderful—a couple edits and then this should be ready to go. If you're still having issues with `devtools::document()` like those noted in the linked issue, feel free to just ask us to do it on your behalf!

Much appreciated, Luis!

**Comment by luisDVA** [comment]
thanks Simon, I've restored the NEWS.md file from main
